### DISCUSSION TITLE
Marcdoc Image handling in Next.js

### DISCUSSION THREAD

**Discussion by samuelervin** [markdoc/markdoc#532] [Show and tell]
I was having trouble with images in Markdoc not coming up and thought I might share this solution I am using. When writing markdown it is easy to use the ![myimage](./myimage.png) syntax. However it usually renders terrible and you can't size it without changing the images. Generally in those situations I would just use an img tag and move on. However while investigating moving some docs over that wouldn't work with markdoc for me and I wanted the cool image features Next.js provides me. So I added a component and a tag . 

First create the component _DocImage.ts_ in your app/components directory: 
`
import Image from 'next/image';
`
`
export function DocImage({
    title,
    src,
    width,  
    height,
    quality,
    style
  }: {
    title: string
    src: string
    width: number
    height: number
    quality: number
    style: string
  }) {
`
`
    return (
          <Image className={style} width={width} height={height} alt={title} src={src} quality={quality}/>    
    )
  }
`
Then Add it to your markdoc/tags.js inside your const tags:
`
const tags = {
      docimage:{
      attributes:{
        title: {type: String},
        width: {type: Number},
        height: {type: Number},
        src: {type: String},
        quality:{type: Number},
        style:{type: String},
      },
      render: DocImage,
    }
  , ...
` 

Once done you can use it in your markdown files like so:
`
{% docimage title="Openstack Select" height=15 width=15 src="/images/hamburger.png" style="inline" %} {% /docimage %}
`
Because I use VSCode I will add a snippet soon to make it easier while writting docs to add these as I go along quickly. 
Also notice you can send CSS style to the Image component that will allow you to use Tailwindcss or other CSS to get your image lined up correctly in your document. 

I haven't gone down the road of static images yet with it, but I think adding that to the component level is well covered in other docs. Hope it helps someone learning Markdoc.

**Comment by mfix-stripe** [comment]
Yes! This is what we typically suggest people do.

Ideally though, it would be great if we could use the regular Markdown image syntax 
```md
![Openstack Select](/images/hamburger.png)
```
and have Next.js automatically pick up the width and height (which is how `next/image` currently works when using image imports)

Would be additionally could if we could pass the width/height w/ Markdoc syntax somehow

```md
![Openstack Select](/images/hamburger.png) {% width=15 height=15 %}
or
![Openstack Select {% width=15 height=15 %}](/images/hamburger.png)

**Reply by samuelervin** [reply]
That would be a great and make it easier to use without having to remember any additional elements to use.
